src/mscorlib/src/System/TimeZoneInfo.cs

```diff
@@ -1944,10 +1971,173 @@ private class OffsetAndRule {
+        /// 1. Read the TZ environment variable.  If it is set, use it.
+        /// 2. Look for the data in /etc/localtime.
+        /// 3. Look for the data in c_timeZoneDirectory/localtime.
+        /// 4. Use UTC if all else fails.
+        /// </summary>
         [SecurityCritical]
-        internal static Byte[] GetLocalTzFile() {
-            // UNIXTODO
-            return null;
+        private static bool TryGetLocalTzFile(out byte[] rawData, out string id)
+        {
+            id = null;
+            string tzFilePath = null;
+            string tzVariable = GetTzEnvironmentVariable();
+
+            // GetTzEnvironmentVariable() will return non-null if the variable was set.
+            // In the case where it is set to an empty string, UTC should be used - i.e. TryGetLocalTzFile() should return false.
+
+            if (tzVariable != null)
+            {
+                if (tzVariable.Length > 0)
+                {
+                    // if tzVariable isn't an absolute path, make it relative to the system timezone directory
+                    if (tzVariable[0] != '/')
+                    {
+                        id = tzVariable;
+                        tzFilePath = Path.Combine(c_timeZoneDirectory, tzVariable);
+                    }
+                    else
+                    {
+                        tzFilePath = tzVariable;
+                    }
+
+                    if (!File.Exists(tzFilePath))
+                    {
+                        rawData = null;
+                        id = null;
+                        return false;
+                    }
+                }
+            }
+            else
+            {
+                const string localtimeFilePath = "/etc/localtime";
+                if (File.Exists(localtimeFilePath))
+                {
+                    // UNIXTODO: optimzation - see if localtimeFilePath is a symlink, and use 'readlink' to get the id
+                    tzFilePath = localtimeFilePath;
+                }
+
+                if (tzFilePath == null)
+                {
+                    string usrLocalTimeFilePath = Path.Combine(c_timeZoneDirectory, "localtime");
+                    if (File.Exists(usrLocalTimeFilePath))
+                    {
+                        tzFilePath = usrLocalTimeFilePath;
+                    }
+                }
+            }
+
+            if (string.IsNullOrEmpty(tzFilePath))
+            {
+                rawData = null;
+                id = null;
+                return false;
+            }
+
+            try
+            {
+                rawData = File.ReadAllBytes(tzFilePath);
+                if (rawData != null && string.IsNullOrEmpty(id))
+                {
+                    id = FindTimeZoneId(rawData);
+                }
+
+                return rawData != null;
+            }
+            catch (IOException)
+            {
+                rawData = null;
+                id = null;
+                return false;
```

Comment:
This is the same code that happens just above for the null or empty case.  Seems like you could consolidate by moving this to after the catch block (leaving the catch empty) and then wrap the whole try/catch in a !string.IsNullOrEmpty check.